### PR TITLE
:memo: Overhaul the documentation

### PR DESCRIPTION
**Pull request by BattleCh1cken**
#### Summary

This pull request migrates the documentation from a `.pdf` file, which is hard to navigate for users, to a website rendered by `mdbook`.

#### Checklist

- [x] I have fully tested all of the added features
- [x] I have fully documented all of the relevant code

### CODE REVIEW

**Comment by BattleCh1cken** [comment]
This pr is blocked by #38 to avoid merge conflicts.

File: docs/src/developer_documentation/project_architecture.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+- `lib.typ`: The entrypoint for the whole template.
+- `internals.typ`: All of the internal function calls that should not be used by
+  theme authors or users.
+- `entries.typ`: Contains the user facing API for entries, as well as the internal
+  template functions for printing out the entries and cover.
+- `glossary.typ`: Contains the user facing API for the glossary.
+- `globals.typ`: Contains all of the global variables for the entire project.
+- `utils.typ`: Utility functions intended to help implement themes.
+- `themes/`: The folder containing all of the themes.
+  - `themes.typ`: An index of all the themes are contained in the template
```
Comment (dblaufuss) [inline docs/src/developer_documentation/project_architecture.md:25]: remove tab

Comment (Isaiah15414) [reply docs/src/developer_documentation/project_architecture.md:25]: it has a tab because `themes.typ` is inside the `themes/` folder

**Review by Isaiah15414 — APPROVED** [review]
Looks great! Just fixed some typos, but everything reads very well

File: docs/src/developer_documentation/project_architecture.md (hunk 1 above)
Comment (dblaufuss) [reply docs/src/developer_documentation/project_architecture.md:25]: I realize my mdbook was out of date.

File: docs/src/installation.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+The only thing you'll need in preparation is a [GitHub](https://github.com/) account (only if using Codespaces) and [VSCode](https://code.visualstudio.com/) installed. Once you do that, just follow the instructions in the [README](https://github.com/The-Notebookinator/quick-start-template).
+
+## Local Installation
+
+This installation process is a little harder, and requires more software to be manually installed on your computer.
+
+Make sure you have the following software installed:
+
+- [Typst](https://github.com/casey/just#installation)
```
Comment (dblaufuss) [inline docs/src/installation.md]: fix this hyperlink

Comment (dblaufuss) [reply docs/src/installation.md]: git too actually

Comment (BestUsernamEver) [reply docs/src/installation.md:24]: What's wrong with it?

**Pull request merged**